### PR TITLE
Record refactor

### PR DESCRIPTION
**Pull request by tokebe**
Refactors code to support the record refactor.

Requires the following additional PRs:
- biothings/api-respone-transform.js/pull/29
- biothings/call-apis.js/pull/50

**IMPORTANT:**

This PR will break any currently-present cached records, but shouldn't break existing completed async jobs that have had their results cached.

Additionally, this PR:
- Updates the `CacheHandler` to leverage new Record implementation for greater space efficiency
- Fixes the skipped queries count to be accurate (not include first failed query)
- Fixes an edge case issue where the scoring function might cause a query to fail

### CODE REVIEW

**Comment by tokebe** [comment]
Noting that all currently failing tests are failing for known reasons related to automated github testing and should all pass on local.

File: src/query_results.js [hunk 1]
```diff
@@ -48,7 +39,7 @@ const { getScores, calculateScore } = require('./score');
  */
 module.exports = class TrapiResultsAssembler {
   /**
-   * Create a QueryResult instance.
+   * Create a QueryResult i9nstance.
```
Comment (ariutta) [inline src/query_results.js:42]: minor typo

**Comment by ariutta** [comment]
Is there a way to move the [jsdoc annotation for `Record`](https://github.com/biothings/bte_trapi_query_graph_handler/blob/record-refactor/src/query_results.js#L7-L32) to [record.ts](https://github.com/biothings/api-respone-transform.js/blob/record-refactor/src/record.ts) and import it into query_results.js? I know you can import TypeScript definitions, but I don't know about JSDoc annotations.

**Comment by ariutta** [comment]
Overall, this looks great! I've only read the code, not done actual testing. It'll be important to ensure we test edge cases about directionality with the switch of input/output subject/object.
